### ISSUE TITLE
Panic when given tex files

### ISSUE THREAD

**Issue by lise-henry** [trishume/syntect#30]
Syntect seems to panic when it is given (la)tex files.

# Example

Using `cargo run --example syncat test.tex` with `test.tex` being:

```
\title{}
``` 

This gives the error: 

```
thread 'main' panicked at 'Can only iterate patterns after linking: Named("$top_level_main")', src/parsing/syntax_definition.rs:136
```
(Removing the curly braces, i.e. if the file only contains `\title`, removes this error).

**Comment by trishume** [comment]
I can reproduce this and it's a real bug. I'll try and look into this soon. Thanks for reporting it.

I seem to have accidentally fixed it already in #27 , but I haven't quite finished that branch and Ruby and PHP don't work in it yet. I'll still try and figure out why it broke and why I fixed it though.

**Comment by trishume** [comment]
Turns out LaTeX relies on an obscure feature that I didn't support. Now I half-support it so it shouldn't crash, but I don't think LaTeX will highlight perfectly either until I get #27 shipped.
